internal/services/tenant_isolation_policy/resource_tenant_isolation_policy_test.go

```diff
@@ -0,0 +1,605 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package tenant_isolation_policy_test
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/jarcoal/httpmock"
+	"github.com/microsoft/terraform-provider-power-platform/internal/mocks"
+)
+
+const testTenantID = "00000000-0000-0000-0000-000000000001"
+
+func setupTenantHttpMocks() {
+	// Mock tenant endpoint that's called before CRUD operations.
+	httpmock.RegisterResponder("GET", "https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/tenant?api-version=2021-04-01",
+		func(req *http.Request) (*http.Response, error) {
+			return httpmock.NewStringResponse(http.StatusOK, fmt.Sprintf(`{
+                "id": "/providers/Microsoft.BusinessAppPlatform/tenant",
```

Comment:
RegisterResponder method should be written in as few file lines as possible. Making each Sprintf parameter makes it unreadble. Apply that suggestion to all RegisterResponder calls in this file

Reply:
this is pretty readable to me.